addons/hr_skills/models/hr_skill_type.py

```diff
@@ -20,3 +21,18 @@ def _set_default_level(self):
         for types in self:
             if not types.skill_level_ids.filtered('default_level'):
                 types.skill_level_ids[:1].default_level = True
+
+    def write(self, vals):
+        if 'active' in vals and not vals['active']:
+            self.env['hr.skill'].search([('skill_type_id', 'in', self.ids)]).write({'active': False})
+            self.env['hr.skill.level'].search([('skill_type_id', 'in', self.ids)]).write({'active': False})
+        return super().write(vals)
+
+    def toggle_active(self):
+        res = super().toggle_active()
+        skill_type_active = self.filtered('active')
+        skills = self.env['hr.skill'].with_context(active_test=False).search([('skill_type_id', 'in', skill_type_active.ids)])
+        skills.action_unarchive()
+        skill_levels = self.env['hr.skill.level'].with_context(active_test=False).search([('skill_type_id', 'in', skill_type_active.ids)])
+        skill_levels.action_unarchive()
+        return res
```

Comment:
Why does the write only archive related content while toggle_active only unarchives it? Wouldn't it make more sense to only override `toggle_active` and toggle skills and skill levels accordingly?
Also, it feels overkill to use search to fetch skills and skill levels are both are stored respectively in `skill_ids` and `skill_level_ids` fields.
```suggestion
    def toggle_active(self):
        res = super().toggle_active()
        active_skill_type = self.filtered('active')
        active_skill_type.skill_ids.action_unarchive()
        active_skill_type.skill_level_ids.action_unarchive()
        (self - active_skill_type).skill_ids.action_archive()
        (self - active_skill_type).skill_level_ids.action_archive()
        return res
```

Reply:
> Why does the write only archive related content while toggle_active only unarchives it? Wouldn't it make more sense to only override `toggle_active` and toggle skills and skill levels accordingly? Also, it feels overkill to use search to fetch skills and skill levels are both are stored respectively in `skill_ids` and `skill_level_ids` fields.

When we unarchive the skill type then skill and skill level are not set. so i did above code. what do you think ?

Reply:
Hello @Bertrand2
  
FYI: I have taken the reference from the helpdesk stage [ref](https://github.com/odoo/enterprise/blob/master/helpdesk/models/helpdesk_stage.py#L50-L57)